src/tensor_ops/conv.rs

```diff
@@ -55,10 +62,27 @@ impl<const B: usize, const C: usize, const H: usize, const W: usize, T: Tape>
         let phb = bias.phantom();
         let phr = result.phantom();
         tape.add_backward_op(move |grads| {
-            let (fg, bg, ig, r_grad) = grads.muts_and_ref(&phf, &phb, &x, &phr);
-            let f = f.data();
-            for ((x_i, rg_i), ig_i) in x.data().iter().zip(r_grad.iter()).zip(ig.iter_mut()) {
-                <Cpu as DeviceConv2D<S, P>>::conv_backward(x_i, f, rg_i, ig_i, fg, bg);
+            let (fg, bg, ig, rg) = grads.muts_and_ref(&phf, &phb, &x, &phr);
+
+            let f_data = f.data();
+            let mut fs: Box<[[[[[f32; K]; K]; C]; O]; B]> = Cpu::zeros();
+            let mut bs: Box<[[f32; O]; B]> = Cpu::zeros();
```

Comment:
Maybe there is an alternative parallelization scheme that doesn't require allocating these two temporary buffers. Unsure how else to address multiple threads updating the filters/bias at the same time.